common/interfaces/desktop_app_interface/usb_event.c

```diff
@@ -91,14 +95,54 @@ static usb_event_t usb_event;
  * STATIC FUNCTION PROTOTYPES
  *****************************************************************************/
 
+/**
+ * @brief Validates that the received usb event is valid in the current context
+ * of the application/OS.
+ *
+ * @param evt The captured usb event usb_event_t
+ */
+usb_evt_error_t validate_usb_event(usb_event_t *evt);
+
+/**
+ * @brief Reset the members of usb_event_t to decided default state.
+ * @details The instance after reset will represent the absence of any usb
+ * event as well as the members will be invalid/meaningless content.
+ *
+ * @param evt Pass the reference to an instance of usb_event_t to reset
+ */
+void reset_event_obj(usb_event_t *evt);
+
 /*****************************************************************************
  * STATIC FUNCTIONS
  *****************************************************************************/
 void reset_event_obj(usb_event_t *evt) {
   evt->flag = false;
-  evt->cmd_id = 0;
-  evt->p_msg = NULL;
   evt->msg_size = 0;
+  evt->p_msg = NULL;
+}
+
+usb_evt_error_t validate_usb_event(usb_event_t *evt) {
+  usb_evt_error_t status = USB_EVT_NO_ERROR;
+
+  if (NULL == evt) {
+    return status;
+  }
+
+  core_command_t req_meta = CORE_COMMAND_INIT_ZERO;
+  pb_istream_t stream = pb_istream_from_buffer(request_meta, request_meta_size);
+
+  if (false == pb_decode(&stream, CORE_COMMAND_FIELDS, &req_meta)) {
+    return USB_ERR_INVALID_REQ_META;
+  }
+
+  // TODO: verify if the req_meta.applet_id is valid one ref PR #235
+  // const cy_app_desc_t *app_desc = registry_get_app_desc(req_meta.applet_id);
+
+  // if (NULL == app_desc) {
+  //   return USB_ERR_UNKNOWN_APP;
+  // }
+
```

Comment:
Maybe return NO_ERROR as of now by default

Reply:
added